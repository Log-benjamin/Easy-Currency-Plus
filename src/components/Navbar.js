import { NavLink, useMatch, useResolvedPath } from 'react-router-dom';
import moment from 'moment';

export default function Navbar() {
  const date = moment().format('MMMM DD YYYY');
  return (
    <header>
      <nav className="nav-section">
        <div className="disp-date">
          <ul>
            <CustomeLink to="/">
              <i className="fa-solid fa-circle-chevron-left fa-xl" />
            </CustomeLink>
            <h3>
              {date}
            </h3>
          </ul>
        </div>
        <ul className="micandsetting">
          <CustomeLink to="/micsearch"><i className="fa-solid fa-microphone-lines fa-lg" /></CustomeLink>
          <CustomeLink to="/setting"><i className="fa-solid fa-gear fa-lg" /></CustomeLink>
        </ul>
      </nav>
    </header>
  );
}

function CustomeLink({ to, children }) {
  const resolvedPath = useResolvedPath(to);
  const isActive = useMatch({ path: resolvedPath.pathname, end: true });

  return (
    <li className={isActive ? 'active' : ''}>
      <NavLink to={to}>{children}</NavLink>
    </li>
  );
}
