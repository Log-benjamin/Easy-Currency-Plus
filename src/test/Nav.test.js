import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';

describe('Navbar Should render as expected', () => {
  it('should be similar with the snapshot', () => {
    const output = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );
    expect(output).toMatchSnapshot();
  });
});
