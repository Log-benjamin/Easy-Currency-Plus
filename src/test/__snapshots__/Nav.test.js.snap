// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Should render as expected should be similar with the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <nav
          class="nav-section"
        >
          <div
            class="disp-date"
          >
            <ul>
              <li
                class="active"
              >
                <a
                  aria-current="page"
                  class="active"
                  href="/"
                >
                  <i
                    class="fa-solid fa-circle-chevron-left fa-xl"
                  />
                </a>
              </li>
              <h3>
                August 11 2023
              </h3>
            </ul>
          </div>
          <ul
            class="micandsetting"
          >
            <li
              class=""
            >
              <a
                class=""
                href="/micsearch"
              >
                <i
                  class="fa-solid fa-microphone-lines fa-lg"
                />
              </a>
            </li>
            <li
              class=""
            >
              <a
                class=""
                href="/setting"
              >
                <i
                  class="fa-solid fa-gear fa-lg"
                />
              </a>
            </li>
          </ul>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header>
      <nav
        class="nav-section"
      >
        <div
          class="disp-date"
        >
          <ul>
            <li
              class="active"
            >
              <a
                aria-current="page"
                class="active"
                href="/"
              >
                <i
                  class="fa-solid fa-circle-chevron-left fa-xl"
                />
              </a>
            </li>
            <h3>
              August 11 2023
            </h3>
          </ul>
        </div>
        <ul
          class="micandsetting"
        >
          <li
            class=""
          >
            <a
              class=""
              href="/micsearch"
            >
              <i
                class="fa-solid fa-microphone-lines fa-lg"
              />
            </a>
          </li>
          <li
            class=""
          >
            <a
              class=""
              href="/setting"
            >
              <i
                class="fa-solid fa-gear fa-lg"
              />
            </a>
          </li>
        </ul>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
